test/unit-test/mqtt_agent_utest.c

```diff
@@ -1102,22 +1137,21 @@ void test_MQTTAgent_CommandLoop_with_empty_command_queue( void )
 {
     MQTTStatus_t mqttStatus;
     MQTTAgentContext_t mqttAgentContext;
-    MQTTAgentCommandFuncReturns_t pReturnFlags = { 0 };
 
     setupAgentContext( &mqttAgentContext );
 
-    mqttAgentContext.mqttContext.connectStatus = MQTTConnected;
-    pReturnFlags.addAcknowledgment = false;
-    pReturnFlags.runProcessLoop = true;
-    pReturnFlags.endLoop = true;
+    mqttAgentContext.mqttContext.connectStatus = MQTTNotConnected;
+    returnFlags.addAcknowledgment = false;
+    returnFlags.runProcessLoop = true;
+    returnFlags.endLoop = false;
 
     MQTTAgentCommand_ProcessLoop_ExpectAnyArgsAndReturn( MQTTSuccess );
-    MQTTAgentCommand_ProcessLoop_ReturnThruPtr_pReturnFlags( &pReturnFlags );
-    MQTT_ProcessLoop_IgnoreAndReturn( MQTTSuccess );
+    MQTTAgentCommand_ProcessLoop_ReturnThruPtr_pReturnFlags( &returnFlags );
+    MQTTAgentCommand_ProcessLoop_ExpectAnyArgsAndReturn( MQTTRecvFailed );
```

Comment:
Looking at MQTTAgentCommand_ProcessLoop() it can't return anything except MQTTSuccess: https://github.com/FreeRTOS/coreMQTT-Agent/blob/560ce2cc9c60e6aa1be7d19a62a2d95eb2e00b54/source/mqtt_agent_command_functions.c#L55
We should probably be making sure the values we use in our unit tests are actually possible. If this is for coverage, we should be testing paths where it is actually possible for there to be an error. If we need to write another test case that's specifically for this situation then we can just do so.

Reply:
Added a new test case for coverage.